### PR TITLE
Quarkus Azure storage blob extension

### PR DESCRIPTION
**Pull request by majguo**
The PR relates to issue #2 , which:
- Support to automatically configure a `BlobServiceClient` bean with property `quarkus.azure.storage.blob.connection-string`
- Adds unit tests

The following tasks are not completed yet:
- DevServicesStorageBlobProcesser
- integration test
- docs
- pipelines

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: storage-blob/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit4-mock</artifactId>
```
Comment (melloware) [inline storage-blob/deployment/pom.xml:43]: Shouldnt this be `test` scope?

File: storage-blob/deployment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            <groupId>io.quarkiverse.azureservices</groupId>
+            <artifactId>quarkus-azure-storage-blob</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>commons-logging-jboss-logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
```
Comment (melloware) [inline storage-blob/deployment/pom.xml:33]: Test scope?

File: storage-blob/runtime/src/main/java/io/quarkiverse/azureservices/azure/storage/blob/runtime/StorageBlobConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "azure.storage.blob", phase = ConfigPhase.RUN_TIME)
+public class StorageBlobConfiguration {
+
+    /**
+     * The connection string of Azure Storage Account.
+     */
+    @ConfigItem
+    public String connectionString;
```
Comment (melloware) [inline storage-blob/runtime/src/main/java/io/quarkiverse/azureservices/azure/storage/blob/runtime/StorageBlobConfiguration.java:14]: This really should be 3 separate configuration properties.  Mostly so the account key can be stored in a Vault Secret.

File: storage-blob/deployment/pom.xml (hunk 1 above)
Comment (majguo) [reply storage-blob/deployment/pom.xml:43]: Thanks for your review and sorry for my late response. The dependency declaration is removed as it's already included in `quarkus-devservices-deployment` when I was developing Dev Services for Azure Storage Blob. Details pls see https://github.com/quarkiverse/quarkus-azure-services/pull/4/commits/a83dce3bb62ae17c04ca3eb13c17bc08c695294d.

File: storage-blob/deployment/pom.xml (hunk 2 above)
Comment (majguo) [reply storage-blob/deployment/pom.xml:33]: Same story above, see https://github.com/quarkiverse/quarkus-azure-services/pull/3#discussion_r1009037426.

File: storage-blob/runtime/src/main/java/io/quarkiverse/azureservices/azure/storage/blob/runtime/StorageBlobConfiguration.java (hunk 3 above)
Comment (majguo) [reply storage-blob/runtime/src/main/java/io/quarkiverse/azureservices/azure/storage/blob/runtime/StorageBlobConfiguration.java:14]: Good point. So far I expose `quarkus.azure.storage.blob.connection-string` for the simplicity, will consider more config properties that're supported by `com.azure.storage.blob.BlobServiceClient` in the future, as well as storing account key in a Vault Secret.

**Pull request merged**